lib/src/piece/assign.dart

```diff
@@ -67,181 +62,89 @@ import 'piece.dart';
-  static const State _blockSplitRight = State(2, cost: 0);
-
-  /// Split at the operator.
-  static const State _atOperator = State(3);
-
-  /// The left-hand side of the operation.
-  final Piece? _left;
+  static const State _blockSplitRight = State(1, cost: 0);
 
-  // TODO(rnystrom): If it wasn't for the need to constrain [_left] to split
-  // in [applyConstraints()], we could write the operator into the same piece
-  // as [_left]. In the common case where the AssignPiece is for a named
-  // argument, the name and `:` would then end up in a single atomic
-  // [CodePiece].
+  /// Force the left-hand side to block split and allow the right-hand side to
+  /// split.
+  static const State _blockSplitLeft = State(2);
 
-  /// The `=` or other operator.
-  final Piece _operator;
+  /// The left-hand side of the operation and the operator itself.
+  final Piece _left;
 
   /// The right-hand side of the operation.
   final Piece _right;
 
-  /// If `true`, then the left side supports being block-formatted, like:
+  /// Whether the piece should have a cost for splitting at the operator.
   ///
-  ///     var [
-  ///       element1,
-  ///       element2,
-  ///     ] = value;
-  final bool _canBlockSplitLeft;
-
-  /// If `true` then the right side supports being block-formatted, like:
+  /// Usually true because it's generally better to block split inside the
+  /// operands when possible. But false for `=>` when the expression has a form
+  /// where we'd rather keep the expression itself unsplit as in:
   ///
-  ///     var list = [
-  ///       element1,
-  ///       element2,
+  ///     // Don't avoid split:
+  ///     makeStuff() => [
+  ///       element,
+  ///       element,
   ///     ];
-  final bool _canBlockSplitRight;
-
-  /// If `true` then prefer to split at the operator instead of block splitting
-  /// the right side.
   ///
-  /// This is `true` for `=>` functions whose body is a function call. This
-  /// keeps the called function next to its arguments instead having the
-  /// function name stick to the `=>` while the arguments split. In other words,
-  /// prefer:
-  ///
-  ///     someMethod() =>
-  ///         someFunction(argument, another);
-  ///
-  /// Over:
-  ///
-  ///     someMethod() => someFunction(
-  ///       argument,
-  ///       another,
-  ///     );
-  final bool _avoidBlockSplitRight;
-
-  AssignPiece(
-    this._operator,
-    this._right, {
-    Piece? left,
-    bool canBlockSplitLeft = false,
-    bool canBlockSplitRight = false,
-    bool avoidBlockSplitRight = false,
-  }) : _left = left,
-       _canBlockSplitLeft = canBlockSplitLeft,
-       _canBlockSplitRight = canBlockSplitRight,
-       _avoidBlockSplitRight = avoidBlockSplitRight;
+  ///     // Avoid split:
+  ///     doThing() =>
+  ///       thingToDo(argument, argument);
+  final bool _avoidSplit;
+
+  AssignPiece(this._left, this._right, {bool avoidSplit = true})
+    : _avoidSplit = avoidSplit;
+
+  static int fails = 0;
```

Comment:
Maybe I'm missing something, where is this used?

Reply:
Oops, that was some old debug code. Removed. :)